File: src/main/java/dev/lbuddyboy/flash/util/Callback.java
package dev.lbuddyboy.flash.util;

public interface Callback<T, O> {
    void call(T t, O t2);

    public static interface SingleCallback<T> {
        void call(T t);
    }
    public static interface TripleCallback<T, V, O> {
        void call(T t, V v, O o);
    }
}

File: src/main/java/dev/lbuddyboy/flash/util/Callback.kt
package dev.lbuddyboy.flash.util

interface Callback<T, O> {
    fun call(t: T, t2: O)
    interface SingleCallback<T> {
        fun call(t: T)
    }

    interface TripleCallback<T, V, O> {
        fun call(t: T, v: V, o: O)
    }
}